pilot/pkg/features/experimental.go

```diff
@@ -217,9 +217,25 @@ var (
 	EnableOptimizedServicePush = env.RegisterBoolVar("ISTIO_ENABLE_OPTIMIZED_SERVICE_PUSH", true,
 		"If enabled, Istiod will not push changes on arbitrary annotation change.").Get()
 
-	// This is used in injection templates, it is not unused.
-	EnableNativeSidecars = env.Register("ENABLE_NATIVE_SIDECARS", false,
-		"If set, used Kubernetes native Sidecar container support. Requires SidecarContainer feature flag.")
+	EnableNativeSidecars = func() NativeSidecarMode {
+		v := env.Register("ENABLE_NATIVE_SIDECARS", "false",
+			"If set, used Kubernetes native Sidecar container support. Requires SidecarContainer feature flag."+
+				" Set to true or false to unconditionally enable. Set to auto-beta or auto-stable to automatically enable"+
+				" if support is detected (at the beta or stable level).").Get()
+		switch v {
+		case "never", "false":
+			return NativeSidecarModeNever
+		case "always", "true":
+			return NativeSidecarModeAlways
+		case "auto-beta":
```

Comment:
From my perspective this is not improve UX, on the opposite it is increasing complexity

Reply:
Is it the `auto-beta` vs `auto-stable`? Or even `auto` would be too complex?

Reply:
I mean auto-beta auto-stable, rare people can distinguish them

Reply:
I would rather it just be `never/always/auto` for this PR - whether we use k8s beta features or not should not be a per-feature decision anyway, and should be controlled elsewhere (by `stable` profiles, etc etc).